### ISSUE TITLE
Preview feature feedback: `microsoftSqlServer`

### ISSUE THREAD

**Issue by 2color** [prisma/prisma#4039]
Please share your feedback about the preview of Microsoft SQL Server support that was released in v2.10.0 in this issue.

- If you encounter a bug, please [open a bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=bug_report.md&title=) in this repo.
- If the feature is working well for you, please share this in a comment below or leave a 👍 on this issue.

If you have any questions, don't hesitate to ask them in the #prisma-client channel in the [Prisma Slack](https://slack.prisma.io).

Related epic: https://github.com/prisma/prisma/issues/4081

**Comment by chuanqisun** [comment]
Hi @2color, does this include Azure SQL? or it's just the on prem MS SQL?

**Comment by pimeys** [comment]
Hi @chuanqisun, we do support Azure SQL due to it being SQL Server 2017 underneath. Most of our tests for now run against the dockerized versions of SQL Server 2017 and 2019, but Azure SQL is a first priority for us.

**Comment by josh-hemphill** [comment]
I created a feature request separately in #6823 , but it probably relates to this.

Even thought SQL Server doesn't support Enums, it would be really nice to still be able to write a schema with enums for the intellisense and type information etc. If I ever have to change the database provider it would also be really nice to only have to worry about native type support/conversions and not about how the schema was structured.

Perhaps there could be some sort of flag to enforce the strong enforcement of enum support that if disabled would create a closest analog in SQL Server that at least expresses the intent of the column being an enum to those working directly with the database.

**Comment by luomavaltteri** [comment]
Howdy!
We are using Prisma with Azure SQL in our project and it has worked well this far! 🚀

Two things we noticed:
- When connecting to Azure SQL, you need to escape at least the user in the connection string (because of the "@" between username and server name). Figuring out how to do that in JDBC connection strings took a bit of time (put the value in curly braces), as that wasn't needed with the SQL server we used locally for development.
- Compared to migration scripts Entity Framework generates, Prisma's migration scripts don't wrap them in transactions or track the migration history in the scripts themselves. I'm not an expert on the topic, so I'm not sure if these things are really needed, but maybe a thing to look at? Below a sample of the EF migration script

```
BEGIN TRANSACTION;
GO

<migrate stuff>

INSERT INTO [__EFMigrationsHistory] ([MigrationId], [ProductVersion])
VALUES (N'20XXXXXXXXXX_AddObject', N'5.0.0');
GO

COMMIT;
GO
```

Keep up the good work 👍

**Comment by pimeys** [comment]
Hey @luomavaltteri,

- Could you give an example of a connection string that works differently in SQL Server compared to Prisma?
- Yeah, due to none of the other databases we support allow running migrations in transactions, we don't do that to SQL Server either. I'm raising up this issue in today's Migrate team meeting though.

**Comment by luomavaltteri** [comment]
So for a SQL server something like this works in the .env file:

```
DATABASE_URL=sqlserver://localhost:1433;database=db;user=sa;password=veryStrongPassword;trustServerCertificate=true;
```

But with Azure SQL escaping the user is needed (and most likely the password also). That's a JDBC connection string copied from Azure portal for the database.

```
DATABASE_URL=sqlserver://<server>.database.windows.net:1433;database=db;user={user@<server>};password={<password>};encrypt=true;trustServerCertificate=false;hostNameInCertificate=*.database.windows.net;loginTimeout=30;
```
The logs prisma CLI gives is a bit odd, but seems to work:

```
❯ npx prisma migrate deploy
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": SQL Server database "undefined" at "<server>};password={<password>};encrypt=true;trustServerCertificate=false;hostNameInCertificate=*.database.windows.net;loginTimeout=30;:1433"

1 migration found in prisma/migrations

The following migration have been applied:

migrations/
  └─ 20210614080427_init/
    └─ migration.sql
      
All migrations have been successfully applied.
```

**Comment by pimeys** [comment]
Checking the connection string problem right away.

Created an issue for you for migrations wrapped in plast... transactions.

**Comment by luomavaltteri** [comment]
Thanks @pimeys, I'll join your Slack 👍

**Comment by Jolg42** [comment]
Thank you @luomavaltteri for the report.
This full connection string (which includes password)
>Datasource "db": SQL Server database "undefined" at "<server>};password={<password>};encrypt=true;trustServerCertificate=false;hostNameInCertificate=*.database.windows.net;loginTimeout=30;:1433"

This is not expected from the CLI, we make sure that only the server host and port are logged for the other connectors and it seems we something for SQL Server.

I fixed that in our dev version and will be part of 2.25.0 release.

**Comment by janconc** [comment]
Hi guys,

Not sure if this question belongs here, but anyway:
I am currently using TypeORM and evaluating Prisma. I have a Database in **Azure/SQL Server** with multiple schemas, each schema having few tables in them. Is it possible to specify for each model in Prisma schema to which DB schema it belongs? In the future I may need to write queries that SELECTs data from tables from multiple Db schemas.

It seems that Prisma always uses the default `dbo` DB schema and I don't see anywhere a possibility to override it.

Keep up the good work!

**Comment by pimeys** [comment]
@janconc Not yet, please follow the [issue](https://github.com/prisma/prisma/issues/1122) for more details.

I wrote a few proposals internally how we could do that, but there is not yet any timeframe for this.

**Comment by janpio** [comment]
For SQL Server the usage of "synonyms" might offer a temporary workaround as well @janconc: https://github.com/prisma/prisma/issues/1122#issuecomment-825774657

**Comment by 2color** [comment]
Closing this now that this has been released to General Availability in the [3.0.1 release](https://github.com/prisma/prisma/releases/tag/3.0.1)
